lib/lexer.js

```diff
@@ -142,15 +144,20 @@ Lexer.prototype._output = function() {
 Lexer.prototype._embed = function() {
   var tag = this._next() + this._next()
     , ch = this._peek();
+
   if(tag == '<' + this._embedChar) {
     this._inEmbed = true;
     if(ch == '=' || ch == '-') return this._token(Token.EMBED, this._next());
     else return this.nextToken();
   } else if(tag == this._embedChar + '>') {
     this._inEmbed = false;
     return this.nextToken();
+  } else if(tag + ch == "-" + this._embedChar + '>') {
+    this._next();
```

Comment:
What if the next character isn't a newline? I believe the `-%>` only strips trailing newlines or maybe whitespace. Should really have tests for each of the cases, but I hadn't set up testing in the project. :(